### PR TITLE
Make likes() work on stmts

### PR DESCRIPTION
**Pull request by mahaloz**
- usability updates to likes() function on statements and a Expression

### CODE REVIEW

File: ailment/expression.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def has_atom(self, atom, identity=True):
             return self.likes(atom)
 
     def __eq__(self, other):
-        return self.likes(other) and self.idx == other.idx
+        return isinstance(other, Expression) and self.idx == other.idx
```
Comment (ltfish) [inline ailment/expression.py:32]: Why?

Comment (mahaloz) [reply ailment/expression.py:32]: @ltfish I need to find the example again. This caused an infinite loop when doing `likes` on a call

Comment (mahaloz) [reply ailment/expression.py:32]: @ltfish why would it make sense to have a `likes` inside an `__eq__`, does this not destroy the point of `likes`?

Comment (mahaloz) [reply ailment/expression.py]: Alright, @ltfish, nvm. I found the loop. I'm just an idiot.

**Comment by ltfish** [comment]
Merged. Thanks!

**Pull request merged**